Fix error occurring when attaching a file to a Task

ChristianAE opened the pull request:
https://github.com/thewhitetulip/Tasks/issues/18 Extend the SQL storing the information to a file by username and creation date.

The SQL storing the information about an attached file was missing the values for the _user_id_ and _creation_date_.

This pull request extends the SQL by these two values.

ChristianAE committed:
Fix error occurring when attaching a file to a Task

Extend the SQL storing the information to a file by username and creation date.

ChristianAE committed:
Fix error occurring when attaching a file to a Task

Extend the SQL storing the information to a file by username and creation date.

ChristianAE commented:
Hello Suraj Patil,

this is my first ever pull request on GitHub. If there is any error or bad practice please point it out to me. I'll fix it as soon as possible.

Kind regards
Christian

thewhitetulip commented:
Thank you for your PR!

I am glad that you begun contributing to open source projects!!

The only error or bad practice for Pull requests is that the description isn't correct.

My first ever Pull Request was to github.com/spyder/spyder and it is still not merged :smile: 

Please find my comment on the line. The thing is for the file table, we load the user_id and not the username. If you get time, don't send the username to the addFile function, get the userID. It won't run!

thewhitetulip commented on db/files.go:
```diff
@@ -11,8 +11,8 @@ import (
-func AddFile(fileName, token string) error {
-	err := taskQuery("insert into files values(?,?)", fileName, token)
+func AddFile(fileName, token, username string) error {
+	err := taskQuery("insert into files values(?,?,?,datetime())", fileName, token, username)
```
```
userID, err := GetUserID(username)
    if err != nil {
        return nil
    }
err := taskQuery("insert into files values(?,?,?,datetime())", fileName, token, userID)
```

For your files table, we have a referential integrity with the user table and we fetch the userID from a function in database called GetUserID. This is what we pass in the taskQuery function otherwise it'll throw an error :-)

If you want to learn Go and contribute more, I am open to get PRs. Keep contributing!

Note: By the way, you might want to delete this Pull Request and do it again, if you are still willing. I did one more commit to the repo, had a bug which didn't allow tasks with null categories in the main feed.

Please let me know of your intentions! Also if you are going to submit a PR, please test it once on your machine after doing the changes I commented in my note on the files.go file

Thanks!

ChristianAE committed:
Merge remote-tracking branch 'origin/master'

; Conflicts:
;	db/files.go

ChristianAE committed:
Fix error occurring when attaching a file to a Task

Extend the SQL storing the information to a file by username and creation date.

ChristianAE commented:
Thanks for your reply. 

I altered the code as suggested, now using the ID of the user.

The only thing different to your suggestions is, that I'm not returning _nil_ when an error occurs. I return the error. The calling method already handles this.

Before I open a new PR please let me know if you have any more remarks.

Kind regards
Christian

thewhitetulip commented:
Thank you! This is great :+1: 

I don't think we need a new PR, this should do fine.

Also it was a mistake from my end to return a nil, I have no idea why I did that :smile:
